addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -64,6 +71,40 @@ options.registry.mailing_list_subscribe = options.Class.extend({
             self.getParent()._onRemoveClick($.Event( "click" ));
         });
     },
+    /**
+     * @override
+     */
+    cleanForSave() {
+        if (!this.$target[0].matches('.o_newsletter_popup')) {
+            this.$target[0].querySelector('.js_subscribe_btn').classList.remove('d-none');
+            this.$target[0].querySelector('.js_subscribed_btn').classList.add('d-none');
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeWidgetState: function (methodName, params) {
+        if (methodName === 'toggleShowThanks') {
+            const subscribeBtnEl = this.$target[0].querySelector('.js_subscribe_btn');
+            return subscribeBtnEl && subscribeBtnEl.classList.contains('d-none');
```

Comment:
I don't like the implicit cast, _computeWidgetState is expected to return a string, return a string